src/Kinto.elm

```diff
@@ -272,262 +282,307 @@ extractKintoError statusCode statusMsg body =
-    Config
+    (Result Error Decode.Value -> msg)
+    -> Config
     -> BucketName
     -> CollectionName
     -> Body
-    -> (Result Error Decode.Value -> msg)
     -> Cmd msg
-replaceCollection config bucket collection body toMsg =
+replaceCollection toMsg config bucket collection body =
     performQuery
+        toMsg
         (KintoRequestWithBody
             config
             (CollectionEndpoint bucket collection)
             "PUT"
             body
         )
-        toMsg
 
 
 replaceRecord :
-    Config
+    (Result Error Decode.Value -> msg)
+    -> Config
     -> BucketName
     -> CollectionName
     -> RecordId
     -> Body
-    -> (Result Error Decode.Value -> msg)
     -> Cmd msg
-replaceRecord config bucket collection recordId body toMsg =
+replaceRecord toMsg config bucket collection recordId body =
     performQuery
+        toMsg
         (KintoRequestWithBody
             config
             (RecordEndpoint bucket collection recordId)
             "PUT"
             body
         )
-        toMsg
 
 
 
 -- DELETE
 
 
-deleteBucket : Config -> BucketName -> (Result Error Decode.Value -> msg) -> Cmd msg
-deleteBucket config bucket toMsg =
+deleteBucket : (Result Error Decode.Value -> msg) -> Config -> BucketName -> Cmd msg
+deleteBucket toMsg config bucket =
     performQuery
+        toMsg
         (KintoRequest
             config
             (BucketEndpoint bucket)
             "DELETE"
         )
-        toMsg
 
 
 deleteCollection :
-    Config
+    (Result Error Decode.Value -> msg)
+    -> Config
     -> BucketName
     -> CollectionName
-    -> (Result Error Decode.Value -> msg)
     -> Cmd msg
-deleteCollection config bucket collection toMsg =
+deleteCollection toMsg config bucket collection =
     performQuery
+        toMsg
         (KintoRequest
             config
             (CollectionEndpoint bucket collection)
             "DELETE"
         )
-        toMsg
 
 
 deleteRecord :
-    Config
+    (Result Error Decode.Value -> msg)
+    -> Config
     -> BucketName
     -> CollectionName
     -> RecordId
-    -> (Result Error Decode.Value -> msg)
     -> Cmd msg
-deleteRecord config bucket collection recordId toMsg =
+deleteRecord toMsg config bucket collection recordId =
     performQuery
+        toMsg
         (KintoRequest
             config
             (RecordEndpoint bucket collection recordId)
             "DELETE"
         )
-        toMsg
+
+
+
+-- New API experiment
+
+
+send : (Result Error a -> msg) -> HttpBuilder.RequestBuilder a -> Cmd msg
+send tagger builder =
+    builder
+        |> HttpBuilder.send (toKintoResponse >> tagger)
+
+
+get : RecordId -> ResourceConfig a -> HttpBuilder.RequestBuilder a
+get recordId config =
+    RecordEndpoint config.bucket config.collection recordId
+        |> endpointUrl config.baseUrl
+        |> HttpBuilder.get
+        |> HttpBuilder.withHeaders config.headers
+        |> HttpBuilder.withExpect (Http.expectJson config.recordDecoder)
```

Comment:
Here's an idea: maybe we can push this a little bit further. Let's have the requests include something to reflect the kind of request they're making (and thus, the kind of response they will get). So, how about having `get` produce an `HttpBuilder.RequestBuilder (GetResponse a)`? This could include permissions or whatever other non-record stuff is in a GET response. Then, you could add a `Kinto.justRecord : HttpBuilder.RequestBuilder (GetResponse a) -> HttpBuilder.RequestBuilder a`. I think this will also make it harder to accidentally tag a CreateRecord request with a DeleteRecord message type. What do you think? Too much?